### ISSUE TITLE
log4cxx: No appender could be found for logger (BtCommon). Unable to initialize Davinci Resolve

### ISSUE THREAD

**Issue by utherbone** [leinardi/JDInstaller#19]
Happy New Year! @leinardi 

System: ZorinOS 17.2 (UBUNTU 22.04LTS)
GPU: XFX RX590

Hi, Your makefile wasn't working properly, so I looked for the packages it wasn't finding and it was just their names that changed in Ubuntu 24.04 LTS.

I changed the package names in the file "`roles/davinci_resolve/tasks/main.yml`" from:
```
      - libapr1t64
      - libaprutil1t64
      - libasound2t64
      - libglib2.0-0t64
```

to:
```
      - libapr1
      - libaprutil1
      - libasound2
      - libglib2.0-0
```
It worked fine and installed completely.

I decided to start it and after the welcome screen Davinci Resolve doesn't start anymore, so I decided to try to start it through the terminal, and I came across this message:

$ /opt/davinci-resolve/bin/resolve
ActCCMessage Already in Table: Code= c005, Mode= 13, Level=  1, CmdKey= -1, Option= 0
ActCCMessage Already in Table: Code= c006, Mode= 13, Level=  1, CmdKey= -1, Option= 0
ActCCMessage Already in Table: Code= c007, Mode= 13, Level=  1, CmdKey= -1, Option= 0
ActCCMessage Already in Table: Code= 2282, Mode=  0, Level=  0, CmdKey= 8, Option= 0
log4cxx: No appender could be found for logger (BtCommon).
log4cxx: Please initialize the log4cxx system properly.
19.1.2.0003 Linux/Clang x86_64
Main thread starts: DED1D000
log4cxx: setFile(./logs/rollinglog.txt,true) call failed.
log4cxx: IO Exception : status code = 13
0x725bded1d000 | Undefined            | INFO  | 2024-12-30 16:39:04,749 | --------------------------------------------------------------------------------
log4cxx: No output stream or file set for the appender named [RollLogAppender].
0x725bded1d000 | Undefined            | INFO  | 2024-12-30 16:39:04,749 | Loaded log config from /home/utherbone/.local/share/DaVinciResolve/configs/log-conf.xml
0x725bded1d000 | Undefined            | INFO  | 2024-12-30 16:39:04,749 | --------------------------------------------------------------------------------
Segmentation falt

**Comment by leinardi** [comment]
Happy New Year, @utherbone! 🎉  

Thank you for providing detailed feedback. 

To clarify, the script in this repository is specifically designed for **Ubuntu 24.04 LTS**. Using it with other versions of Ubuntu or derivative distributions, like Zorin OS, may lead to missing dependencies or misconfigurations. I recommend verifying the OS compatibility before using the script to avoid such issues. 

Regarding the **DaVinci Resolve segmentation fault**, unless some dependencies are missing, the issue you are experiencing seems to be related to the application itself rather than the setup script. The best course of action would be to contact the official DaVinci Resolve support team for a more precise diagnosis.  

That said, here are a few steps you can try to debug the issue:  

### Debugging Steps:
1. **Check for Missing Dependencies:**
   - Run:  
     ```bash
     ldd /opt/davinci-resolve/bin/resolve
     ```
     Look for any "not found" entries. These indicate missing libraries that need to be installed.

2. **Verify Permissions:**
   - Ensure the user running DaVinci Resolve has read/write access to its directories:
     ```bash
     chmod -R u+rw ~/.local/share/DaVinciResolve
     chmod -R u+rw /opt/davinci-resolve
     ```

3. **Install Missing Dependencies:**
   - Based on your GPU (RX590), ensure you have the latest **Mesa drivers** and **Vulkan support** installed:
     ```bash
     sudo apt install mesa-vulkan-drivers mesa-opencl-icd
     ```

4. **Log File Issues:**
   - The errors related to `rollinglog.txt` and log configuration suggest a permissions or path issue. Ensure the log directory exists and is writable:
     ```bash
     mkdir -p ~/.local/share/DaVinciResolve/logs
     chmod -R u+rw ~/.local/share/DaVinciResolve/logs
     ```

5. **Use OpenCL Instead of CUDA:**
   - If you're using Mesa drivers, make sure OpenCL is enabled for DaVinci Resolve:
     ```bash
     sudo apt install clinfo
     clinfo  # Verify that OpenCL is recognized
     ```

6. **Run with Debugging Tools:**
   - Use `gdb` to get more details about the segmentation fault:
     ```bash
     gdb /opt/davinci-resolve/bin/resolve
     run
     ```
     This can help identify the root cause of the crash.

If these steps don't resolve the issue, it’s likely a deeper compatibility or application-specific bug that requires attention from Blackmagic Design.

BTW, since your distribution is based on Ubuntu 22.04, you might want to try following these [[alternative instructions](https://www.youtube.com/watch?v=d0I-d7Cht3o)](https://www.youtube.com/watch?v=d0I-d7Cht3o), which are specifically tailored for Ubuntu 22.04 and could help resolve the issues you're facing.

Let me know how it goes.

**Comment by utherbone** [comment]
Hi @leinardi it was a problem when it came to whether my ZorinOS 17.2/Ubuntu 22. 04 had OpenCL support, the segmentation error and the log4cxx unconfigured/not found problem, these problems were related to ROCM's proprietary driver and OpenCL, because I wanted maximum performance for both OpenCL and HIP, using my RX590 GPU, but because I study AI at university and I have a YouTube channel that I'm trying to leverage, so a better editor than Kdenlive (not that Kdenlive is bad, it's just that it works with frames and working with time is better for me) will do just fine.

After testing a lot of combinations, which is why it took me so long to reply, I discovered that ROCM HIP works without the proprietary driver, which is very good.

In short, I discovered that Mesa has evolved a lot and nowadays its OpenCL support is very good. It would be great to see HIP support one day too. Well what about your script, it works very well, and I advise you to make a command to automatically change the name of the packages so that it works with distros based on Ubuntu 22.04 lts.

**Comment by leinardi** [comment]
Thank you for the detailed follow-up and for explaining the issue! I agree that Mesa has made huge strides recently, and it’s great to hear that it worked well for your setup. I’ll consider adding support for older Ubuntu-based distributions in the future, but for now, the script will remain targeted at Ubuntu 24.04 as the minimum as I don't have time to implement the changes and do all the tests 🙈
